### PR TITLE
Initialization of node. Second take

### LINKED ISSUE #12: Support initializing a Node
```gherkin
Scenario: Initializing a Node
	When I initailize a Node
	Then I should be able to specify the "node mode"
	And I should be able to specify the name of the node
	And I should be able to specify the port
```

### PR DESCRIPTION
**Pull request by i1i1**
Closes #12

This pr copies subspace-desktop node configuration and rewires some of its arguments to builder's one.

### CODE REVIEW

**Review by ozgunozerk — APPROVED** [review]
Looking good to me, you are aware of the clippy warning though, right @i1i1 ?

**Review by achiurizo — APPROVED** [review]

File: src/node.rs [hunk 1]
```diff
@@ -48,28 +76,183 @@ impl Builder {
...
+        let Self {
+            mode: Mode::Full,
+            chain,
+            directory,
+            name,
+        } = self;
+
+        let chain_spec = match chain {
+            Chain::Gemini2a => ConsensusChainSpec::from_json_bytes(
+                include_bytes!("../res/chain-spec-raw-gemini-2a.json").as_ref(),
```
Comment (achiurizo) [inline src/node.rs:95]: nitpick: do we need to [guard the relative path](https://doc.rust-lang.org/std/fs/fn.canonicalize.html) usage here and/or maybe grab the absolute path?

On a separate note, how we are going to handle when [chain specs are updated upstream](https://github.com/subspace/subspace/tree/main/crates/subspace-node/res)?

**Review by achiurizo — APPROVED** [review]

File: src/node.rs (hunk 1 above)
Comment (achiurizo) [reply src/node.rs:95]: Also linking @i1i1 [PR here regards to chainspec management](https://github.com/subspace/subspace/pull/893)

Comment (i1i1) [reply src/node.rs:95]: This thing happens during the compilation, so no need to guard the relative path. Anyway, this code will be removed in next pr.

Comment (nazar-pc) [reply src/node.rs:95]: Chain spec is only updated when we launch a new chain. After that it doesn't change (well, we may edit bootstrap nodes if we need, but that is about it).

**Pull request merged**